### PR TITLE
Advised people to use lldb rather than gdb and provided command ref

### PR DESCRIPTION
**Pull request by yakman2020**
…lation

### CODE REVIEW

File: setup-macosx.md [hunk 1]
```diff
@@ -239,7 +239,13 @@ brew install pkg-config cmake openssl
 
 #### Codesign GDB
 
-Per the Homebrew installation notes, "gdb" requires special privileges
+If practical, it would be better to use the debugger which is provided by
+xcode for debugging purposes. That debugger is much more capable and reliable
```
Comment (jeffaco) [inline setup-macosx.md]: This is normally called `Xcode`, not `xcode`, by Apple.

File: setup-macosx.md [hunk 2]
```diff
@@ -239,7 +239,13 @@ brew install pkg-config cmake openssl
 
 #### Codesign GDB
 
-Per the Homebrew installation notes, "gdb" requires special privileges
+If practical, it would be better to use the debugger which is provided by
+xcode for debugging purposes. That debugger is much more capable and reliable 
+than gdb on the mac.  Developers used to using the gdb debugger can find a
+document with translations between gdb and lldb commands at
+(http://llvm.org/svn/llvm-project/lldb/tags/lldb-168/www/lldb-gdb.html).
```
Comment (jeffaco) [inline setup-macosx.md]: Please format this as a git reference, as per other locations in this document.

File: setup-macosx.md [hunk 3]
```diff
@@ -239,7 +239,13 @@ brew install pkg-config cmake openssl
...
+
+If you must use gdb, per the Homebrew installation notes, "gdb" requires special privileges
 to access machine ports.  You will need to codesign the binary. For
 instructions, see:
 [BuildingOnDarwin](https://sourceware.org/gdb/wiki/BuildingOnDarwin).
```
Comment (jeffaco) [inline setup-macosx.md:251]: Just like this (a reference in git)! 😄

**Review by jeffaco — APPROVED** [review]
Please make referenced changes and commit

File: setup-macosx.md (hunk 2 above)
Comment (yakman2020) [reply setup-macosx.md]: done

File: setup-macosx.md (hunk 1 above)
Comment (yakman2020) [reply setup-macosx.md]: fixed

**Pull request merged**